File: fib.go
package main

import(
	"fmt"
	"strconv"
	"os"
	"time"
)

func fib(x int32) int32{
	if x < 2 {
		return int32(x)
	}else{
		qianVal := 0
		zuoVal := 1
		res := 0
		var i int32;
		for i = 1; i < x; i++{
			res = qianVal+zuoVal
			qianVal = zuoVal
			zuoVal = res
		}
		return int32(res)
	}
}

func doWork(rem int32) int32{
	var acc int32 = 0
	var i int32;
	for i = 0; i < rem; i++{
	    acc += fib((acc+1)%50)
	}
	return acc
}

func main(){
	start := time.Now()
	n, _ := strconv.Atoi(os.Args[1])
	fmt.Printf("%d\n", doWork(int32(n)));
	duration := time.Now().Sub(start).Nanoseconds() / 1000000
	fmt.Printf("LANGUAGE  Go  %v\n", duration)
}

File: fib.cs
using System;

public class fib{
    private static int calcFib(int x){
	if(x < 2){
	    return x;
	}else{
	    int qianVal = 0;
	    int zuoVal = 1;
	    int res = 0;
	    for(int i = 1; i < x; i++){
		res = qianVal+zuoVal;
	        qianVal = zuoVal;
		zuoVal = res;
	    }
	    return res;
	}
    }
    private static int doWork(int rem){
	int acc = 0;
	for(int i = 0; i < rem; i++){
	    acc += calcFib((acc+1)%50);
	}
	return acc;
    }
    public static void Main(string[] args){
	System.Console.WriteLine(doWork(Convert.ToInt32(args[0])));
    }
}
